### DISCUSSION TITLE
How to delay object selection?

### DISCUSSION THREAD

**Discussion by i-prikot** [fabricjs/fabric.js#7179] [Q&A] [answered]
Hi. How to delay object selection?

**Comment by melchiar** [comment, marked as answer]
What do you mean by delay?

**Reply by i-prikot** [reply]
When I select an object on the canvas, I want `setActiveObject` to fire in a few milliseconds. I want to do a mobile touch to select an object and move the canvas. When I start moving the canvas, the object starts moving, but the canvas stays in place.

**Reply by melchiar** [reply]
There's no native way to add a delay to the selection action. I don't think it'd be an easy thing to add since the selection methods are all synchronous and timeouts are async.

Have you tried the `activeOn` property? It makes the selection only occur at `touchend`/`mouseup`

`fabric.Object.prototype.activeOn = "up";`

**Reply by i-prikot** [reply]
Super! Thank you!

**Reply by i-prikot** [reply]
I tried to do like this. But it works poorly on movement. It looks like your solution works 100%.

```
fabric.util.object.extend(fabric.Object.prototype, {
  _delaySelect: true,
  _timerSelect: null,

  clearTimerSelect: function() {
    this._timerSelect && clearTimeout(this._timerSelect);
  },

  delaySelect: function() {
    if (!this._delaySelect) return false;
    
    if (this._timerSelect) {
      this.clearTimerSelect();
    }

    this._timerSelect = setTimeout(() => {
      this.onSelect = function() {}
      this.canvas.setActiveObject(this);
      this.canvas.requestRenderAll();
      this.onSelect = this.delaySelect;
      this.clearTimerSelect();
    }, 250);
    
    return true;
  },

  onDeselect() {    
    this.clearTimerSelect();
  },

  onSelect: function(e) {
    return this.delaySelect(e);
  }
})
```

**Comment by asturur** [comment]
```
fabric.Object.prototype.activeOn = "up";
```

was created exactly for this kind of customization. Great it works as intented
